### PR TITLE
Implementation of savePolicy and loadPolicy

### PR DESCRIPTION
**Pull request by kingiw**
Currently, I only implement the `savePolicy` and `loadPolicy`.

For the CI, I am still figuring out how to deploy a dynamoDB with travisCI.

Another question is who will be responsible for uploading the project to maven repo?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.casbin.adapter</groupId>
+  <artifactId>dynamodb-adapter</artifactId>
+  <version>1.0-SNAPSHOT</version>
```
Comment (nodece) [inline pom.xml:9]: We should use v0.0.1.

File: src/main/java/org/casbin/adapter/DynamoDBAdapter.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    public DynamoDBAdapter(String serviceEndpoint, String signingRegion) {
+        try {
+            this.client = AmazonDynamoDBClientBuilder
+                            .standard()
+                            .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, signingRegion))
+                            .build();
+    
+            this.dynamoDB = new DynamoDB(this.client);
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (nodece) [inline src/main/java/org/casbin/adapter/DynamoDBAdapter.java:75]: Here should throw the error.

File: src/main/java/org/casbin/adapter/DynamoDBAdapter.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        }
+    }
+
+    public void createTable() {
+        try {
```
Comment (nodece) [inline src/main/java/org/casbin/adapter/DynamoDBAdapter.java:80]: Remove `try...catch`. We don't should ignore the error.

File: src/main/java/org/casbin/adapter/DynamoDBAdapter.java [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+            System.out.println("Table is created!");
+        } 
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void dropTable() {
+        this.table = this.dynamoDB.getTable("casbin_rule");
+        try {
```
Comment (nodece) [inline src/main/java/org/casbin/adapter/DynamoDBAdapter.java:98]: same question.

File: src/main/java/org/casbin/adapter/DynamoDBAdapter.java [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                String ptype = entry.getKey();
+                Assertion ast = entry.getValue();
+                for (List<String> rule : ast.policy) {
+                    CasbinRule line = savePolicyLine(ptype, rule);
+                    putCasbinRuleItem(line);
+                }
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (nodece) [inline src/main/java/org/casbin/adapter/DynamoDBAdapter.java:228]: Throw the error.

**Comment by kingiw** [comment]
Some of the codes require try-catch blocks for exception handling. I rewrote those parts as follows:

```java
try {
  // scripts..
} catch (Exception e) {
  e.printstack();
  throw new Error(e);
}
```
Is that okay?

File: pom.xml [hunk 6]
```diff
@@ -6,11 +6,10 @@
 
   <groupId>org.casbin.adapter</groupId>
   <artifactId>dynamodb-adapter</artifactId>
-  <version>1.0-SNAPSHOT</version>
+  <version>v0.0.1</version>
```
Comment (nodece) [inline pom.xml:9]: <version>0.0.1</version>

**Comment by nodece** [comment]
LGTM

**Comment by nodece** [comment]
> Another question is who will be responsible for uploading the project to maven repo?

Hi @hsluoyz , Could you publish it to maven repo?

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.casbin</groupId>
+      <artifactId>jcasbin</artifactId>
+      <version>1.1.0</version>
```
Comment (hsluoyz) [inline pom.xml:30]: Use latest 1.4.0

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.casbin.adapter</groupId>
+  <artifactId>dynamodb-adapter</artifactId>
+  <version>0.0.1</version>
+
+  <name>DynamoDB Adapter for J.Casbin</name>
+  <url>https://github.com/jcasbin/dynamodb-adapter</url>
+
```
Comment (hsluoyz) [inline pom.xml:13]: Add missing tags, refer to: https://github.com/jcasbin/jdbc-adapter/blob/master/pom.xml

```xml
<name>JDBC Adapter for JCasbin</name>
    <description>Load policy from JDBC supported database or save policy to it</description>
    <url>https://github.com/jcasbin/jdbc-adapter</url>
    <inceptionYear>2018</inceptionYear>

    <issueManagement>
        <system>Github</system>
        <url>https://github.com/jcasbin/jdbc-adapter/issues</url>
    </issueManagement>

    <ciManagement>
        <system>Travis CI</system>
        <url>https://travis-ci.org/jcasbin/jdbc-adapter</url>
    </ciManagement>

    <parent>
        <groupId>org.sonatype.oss</groupId>
        <artifactId>oss-parent</artifactId>
        <version>7</version>
    </parent>
    <licenses>
        <license>
            <name>The Apache Software License, Version 2.0</name>
            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
            <distribution>repo</distribution>
        </license>
    </licenses>
    <scm>
        <url>https://github.com/jcasbin/jdbc-adapter</url>
        <connection>git@github.com:jcasbin/jdbc-adapter.git</connection>
        <developerConnection>https://github.com/hsluoyz</developerConnection>
    </scm>
    <developers>
        <developer>
            <name>Yang Luo</name>
            <email>hsluoyz@qq.com</email>
            <url>https://github.com/hsluoyz</url>
        </developer>
    </developers>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>
```

File: src/main/java/org/casbin/adapter/DynamoDBAdapter.java [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+            this.table = this.dynamoDB.createTable(
+                "casbin_rule",
+                Arrays.asList(new KeySchemaElement("ID", KeyType.HASH)),
+                Arrays.asList(new AttributeDefinition("ID", ScalarAttributeType.S)),
+                new ProvisionedThroughput(10L, 10L)
+            );
+            this.table.waitForActive();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (hsluoyz) [inline src/main/java/org/casbin/adapter/DynamoDBAdapter.java:87]: Just throw, no need to print

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.casbin.adapter</groupId>
```
Comment (hsluoyz) [inline pom.xml:7]: Change to: org.casbin

Comment (hsluoyz) [inline pom.xml:7]: Change to: org.casbin

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  <artifactId>dynamodb-adapter</artifactId>
+  <version>0.0.1</version>
+
+  <name>DynamoDB Adapter for J.Casbin</name>
```
Comment (hsluoyz) [inline pom.xml:11]: jCasbin

File: src/test/java/org/casbin/adapter/TestUtils.java [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+                    break;
+                }
+            }
+            if (!updated) {
+                return false;
+            }
+        }
+        return true;
+    }
+}
```
Comment (hsluoyz) [inline src/test/java/org/casbin/adapter/TestUtils.java:72]: New lines for all files.

**Pull request merged**